### PR TITLE
insert torchvision dependency and write tests for cifar10

### PR DESCRIPTION
**Pull request by BaruchG**
## What does this PR do?

Wrote unit test for the CIFAR10 dataset and made the dataset import from torchvision  to reduce code redundancy as outlined in #839.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)
Draft proposal since TrialCifar10 explicitly states that it should not rely on torchvision and this PR makes CIFAR10 a torchvision dependency.

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/datasets/test_datasets.py [hunk 1]
```diff
@@ -1,9 +1,20 @@
 import pytest
 import torch
+
+<<<<<<< HEAD
```
Comment (oke-aditya) [inline tests/datasets/test_datasets.py:4]: Eh it didn't merge cleanly.

Comment (otaj) [reply tests/datasets/test_datasets.py]: Fixed

**Comment by otaj** [comment]
Hi @BaruchG, can I ask why was this closed?

**Comment by BaruchG** [comment]
Sorry @otaj, I was having some issues with my github version and this somehow got deleted.  The branch is now up again at https://github.com/BaruchG/lightning-bolts/tree/cifar until we decide on what to do with hard torchvision dependencies.

**Pull request closed without merging**